### PR TITLE
making configuration and window more extensible

### PR DESCRIPTION
**Pull request by kitchWWW**
Making the configuration and window scripts more extensible by making some functions protected.

### CODE REVIEW

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs [hunk 1]
```diff
@@ -134,7 +139,12 @@ public override void OnInspectorGUI()
 
         if (hasApplicationInfo)
         {
-            selectedToolPanel = GUILayout.Toolbar(selectedToolPanel, toolPanelNames);
+            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken) && WitAuthUtility.AppServerToken.Length == 32)
+            {
+                selectedToolPanel = GUILayout.Toolbar(selectedToolPanel, toolPanelNames);    
+            }else{
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitConfigurationEditor.cs:145]: new line
else
{

**Review by patapizza — APPROVED** [review]

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs [hunk 2]
```diff
@@ -328,14 +338,22 @@ private void DrawApplication(WitApplication application)
...
         {
-            InfoField("Name", application.name);
-            InfoField("ID", application.id);
-            InfoField("Language", application.lang);
-            InfoField("Created", application.createdAt);
-            GUILayout.BeginHorizontal();
-            GUILayout.Label("Private", GUILayout.Width(100));
-            GUILayout.Toggle(application.isPrivate, "");
-            GUILayout.EndHorizontal();
+            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken) && WitAuthUtility.AppServerToken.Length == 32)
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitConfigurationEditor.cs:341]: it's a bit weird to have this here
what do you think @yolanother?

File: Scripts/Runtime/Configuration/WitConfiguration.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public class WitConfiguration : ScriptableObject
         #if UNITY_EDITOR
         public void UpdateData(Action onUpdateComplete = null)
         {
-            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken))
+            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken) && WitAuthUtility.AppServerToken.Length == 32)
```
Comment (kitchWWW) [inline Scripts/Runtime/Configuration/WitConfiguration.cs:38]: Also just calling this out, this is a second behavior change that technically falls into the same boat as the above UI change, though this is additionally saying "don't make server requests if our server token isn't a real token", which could be valuable other ways too.

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs (hunk 2 above)
Comment (yolanother) [reply Scripts/Editor/Configuration/WitConfigurationEditor.cs:343]: Yeah this doesn't feel right. We should probably use a custom property drawer to display application info so we can then override that instead.

File: Scripts/Runtime/Configuration/WitConfiguration.cs (hunk 3 above)
Comment (yolanother) [reply Scripts/Runtime/Configuration/WitConfiguration.cs:38]: You should replace this with WithAuthUtility.IsValidToken. We have one for ServerToken, we should add one for AppServerToken so the logic is all in one place

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs (hunk 2 above)
Comment (kitchWWW) [reply Scripts/Editor/Configuration/WitConfigurationEditor.cs]: addressed

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs [hunk 4]
```diff
@@ -18,6 +19,8 @@
 #endif
 public class WitConfigurationEditor : Editor
 {
+	public IApplicationDrawer appDrawer = new WitApplicationDrawer();
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitConfigurationEditor.cs:22]: fix indent

File: Scripts/Editor/Configuration/WitApplicationDrawer.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public void DrawApplication(WitApplication application)
+    {
+        if (string.IsNullOrEmpty(application.name))
+        {
+            GUILayout.Label("Loading...");
+        }
+        else
+        {
+            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken))
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitApplicationDrawer.cs:24]: do we need to display a message if the condition is true?

File: Scripts/Editor/Configuration/WitConfigurationEditor.cs [hunk 6]
```diff
@@ -134,7 +142,14 @@ public override void OnInspectorGUI()
 
         if (hasApplicationInfo)
         {
-            selectedToolPanel = GUILayout.Toolbar(selectedToolPanel, toolPanelNames);
+            if (!string.IsNullOrEmpty(WitAuthUtility.AppServerToken) && WitAuthUtility.AppServerToken.Length == 32)
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitConfigurationEditor.cs:145]: WitAuthUtility.IsServerTokenValid?

File: Scripts/Editor/Configuration/WitWindow.cs [hunk 7]
```diff
@@ -60,20 +60,22 @@ protected override void OnDrawContent()
             }
         }
 
-        protected override void OnEnable()
-        {
-            WitAuthUtility.InitEditorTokens();
-
+        protected virtual void setWitEditor(){
```
Comment (patapizza) [inline Scripts/Editor/Configuration/WitWindow.cs:63]: { at the next line

Comment (kitchWWW) [reply Scripts/Editor/Configuration/WitWindow.cs:63]: got it, fixed

**Pull request merged**